### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by eliaslevy**
Requires building Hyperscan from source and then pointing `HYPERSCAN_ROOT` to the Hyperscan build or install directory.

Building and executing the examples, `cargo run --example pcapscan` and `cargo run --example  simplegrep`, works after installing WinPcap and after extracting `Packet.lib` and `wpcap.lib` from the WinPcap Developers Pack into a directory and configuring the `LIB` environment variable to point to it.

For some reason I've yet to figure out `cargo test` is failing as the `target\debug\deps\hyperscan-*.exe` executable dies without any useful output.  Any suggestions would be appreciated.

### CODE REVIEW

**Comment by eliaslevy** [comment]
The
```
error[E0599]: no method named `to_primitive_values` found for type `pnet_base::MacAddr` in the current scope
```
error in the build is related to the `pnet` dependencies. It seems two crates depend on different versions of `pnet_base`, which is messing things up.  I fixed it by running `cargo update -p pnet_base:0.23.0 --precise 0.22.0` in the crate root.

**Comment by flier** [comment]
let's me update the dependecies first

File: hyperscan-sys/build.rs [hunk 1]
```diff
@@ -94,9 +96,9 @@ fn main() {
         }
     }
 
-    if cfg!(any(target_os = "macos", target_os = "freebsd")) {
```
Comment (flier) [inline hyperscan-sys/build.rs]: mispell?

Comment (eliaslevy) [reply hyperscan-sys/build.rs]: Typo.  Sorry.  Fixed now.

**Comment by eliaslevy** [comment]
The crashes are caused by the Hyperscan library and Rust using different versions of `malloc` and `free`.  When the library allocates some memory, and then the Rust crate tries to free it using `libc::free`, then the test program crashes.

All tests pass if I insert 
```rest
unsafe { raw::hs_set_allocator(Some(libc::malloc), Some(libc::free)); }
```
into every test.

`hs.dll` seems to the `malloc` and `free` from `ucrtbased.dll` when I compile it on a Windows 10 system using Visual Studio 2019.  Meanwhile the test executable generated by Rust links to the `malloc` and `free` in `api-ms-win-crt-heap-l1-1-0.dll`.

**Comment by eliaslevy** [comment]
I figured out the issue.  CMake in Windows seems to be ignoring `CMAKE_BUILD_TYPE` when building Hyperscan and using the Debug config as the default.  All works when I override this by executing the build doing `MSBuild.exe ALL_BUILD.vcxproj -p:Configuration=Release`.

**Comment by eliaslevy** [comment]
Closing and creating a new rebased PR.

**Pull request closed without merging**